Makefile

```diff
@@ -19,6 +19,9 @@ test-coverage:
 test-unit:
 	$(PYTEST) ./test/unit
 
+test-deprecation:
+	$(PYTEST) -Werror::DeprecationWarning -Werror::FutureWarning ./test/unit
```

Comment:
Can you add this to the actual build pipelines?

Reply:
gonna try, but rusty there, please suggest better ways.

Reply:
As I said, just add the flags to the `Unit` tests below. No point running all the tests again, just to look for deprecation warnings.